util/chunk/disk.go

```diff
@@ -109,15 +124,47 @@ func (l *ListInDisk) Add(chk *Chunk) (err error) {
 		return
 	}
 	l.offsets = append(l.offsets, chk2.getOffsetsOfRows())
-	err = l.bufWriter.Flush()
-	if err == nil {
-		l.diskTracker.Consume(n)
+	l.diskTracker.Consume(n)
+	l.numRowsInDisk += chk.NumRows()
+	return
+}
+
+// AppendRow appends a row to the last chunk in disk, if no chunk has ever been added,
+// a new chunk will be created. Note that the write will be flushed to disk every time.
+func (l *ListInDisk) AppendRow(row Row) (ptr RowPtr, err error) {
+	var format *diskFormatRow
+	if l.disk == nil {
+		l.disk, err = ioutil.TempFile(tmpDir, l.diskTracker.Label().String())
+		if err != nil {
+			return
+		}
+		l.bufWriter = bufWriterPool.Get().(*bufio.Writer)
+		l.bufWriter.Reset(l.disk)
```

Comment:
We can make a method called `initDiskFile` for callings of `AppendRow` and `Add` to avoiding copying code.

Reply:
Good idea, will do.